### DISCUSSION TITLE
How to override explicit interface members

### DISCUSSION THREAD

**Discussion by curt-w** [metalama/Metalama#17] [Q&A] [answered]
I'm trying to understand how to override properties / methods that are explicitly defined within an interface. I'm running into issue with this more complex situation

Provided the following model, how would I write an aspect that is capable of overriding the property getter that is explicitly defined under InterfaceB. For example purposes, let's ".ToUpper()" the result using a ProjectFabric. I haven't had issue targeting the types via. the fabric layer, just having success in the aspect actually being able to manipulate the explicitly defined member within the implemented interface.

```
public interface InterfaceA
{
    string TestStringProperty { get; }
}

public interface InterfaceB : InterfaceA
{
    string InterfaceA.TestStringProperty { get => "base_impl"; }
    string TestStringPropertyLocal { get; }
}

public class TestClass : InterfaceB
{
    
    public string TestStringPropertyLocal { get { return "local_impl"; }}

}
```

**Comment by prochan2** [comment]
Overriding of default interface member implementation is not supported at the moment.

**Reply by curt-w** [reply]
Thanks - is there a feature request for tracking?

My intent with this approach was so my source code would be cognisant that these properties / methods exist and refer to them throughout, while I can worry about the actual implementation within the AOP layer. Not sure if you have any ideas on an alternate solution.

**Reply by prochan2** [reply]
We have filed this feature internally and it is planned to be implemented. We'll let you know as soon as it's released.

We don't see any alternative approach at the moment.

**Comment by gfraiteur** [comment, marked as answer]
The user story has now been implemented.
